src/parser/parser_flow.ml

```diff
@@ -2556,7 +2565,18 @@ end = struct
               body;
             }))
         | _ ->
+            raise Try.Rollback
+      in
+
+      fun env ->
+        let start_loc = Peek.loc env in
+        Expect.token env T_FOR;
+        Expect.token env T_LPAREN;
+        match Try.to_parse env (try_forin_or_forof start_loc) with
```

Comment:
Sure, I can give that a shot. To add some more details to what you've wrote, we should parse the init declaration with implicit_init, so we don't add errors for uninitialized const/destructuring. However, if it turns out that we're in a for loop, we need to re-add those errors by inspecting the parsed AST—that's some duplicate logic I was trying to avoid with backtracking, but it's probably worth it for improved performance.